src/bin/genemichaels.rs

```diff
@@ -244,14 +345,127 @@ fn main() {
                 Ok(_) => { },
                 Err(e) => {
                     if !args.quiet {
-                        eprintln!("Error formatting {}: {:?}", &file.to_string_lossy(), e);
+                        print_error_text();
+                        eprintln!("formatting {}: {:?}", &file.to_string_lossy(), e);
                     }
                     failed = true;
                 },
             };
         }
+        eprintln!(
+            "\x1B[1;32m    Finished\x1B[0;22m workspace formatted successfully in {:.2}s",
+            time::Instant::now().duration_since(inst).as_secs_f64()
+        );
         if failed {
             process::exit(1);
         }
     }
 }
+
+fn process_cargo_toml(path: PathBuf, manifest: Manifest, args: &Args, config: FormatConfig) {
+    let dirs = match manifest.workspace.map(|wkspc| wkspc.members) {
+        Some(mut mb) => {
+            if let Some(bins) = manifest.bin {
```

Comment:
TBH it's been a while since I've used workspaces directly, but I think you'll need to recursively parse workspace Cargo.toml's and process `bin` `lib` etc in each one, adding up the paths.